state/state.md

```diff
@@ -1 +1,45 @@
-# State
+# BlockExecutor
```

Comment:
```suggestion
# Block Executor
```

Reply:
Maybe we should change the name of the file to be `block-executor.md` instead of `state.md`?